### ISSUE TITLE
Need clarification - when you say virtual network do you mean like MiniNet?

### ISSUE THREAD

**Issue by bmullan** [Erik-Lamers1/vnet-manager#23]
> VNet-manager will create and manage the containers you need **_to create a virtual network_**. 

Just to be sure I am not misinterpreting anything... when you say ""**_create a virtual network_**" do you mean a MiniNet type virtual network or a LXD Multi-Host virtual network overlay?

**Comment by Erik-Lamers1** [comment]
VNet-manager currently utilizes both the network on the host itself and the networks used by the LXC containers (which also reside on the host), so it more of containerization then real virtualization. Mininet creates this virtualization by utilizing namespaces, this is currently not implemented for VNet-manager (I didn't see a real use case for this yet). 

It would be possible to create a network overlay within the topology that is created by VNet-manager, but this could with FRR configuration rather then VNet-manager itself.

Basically, VNet-manager makes it easier to hook LXC containers together in self made topologies.

**Comment by bmullan** [comment]
Hi Erik
thanks for the reply!   No, I wasn't looking for a Mininet environment.   I am interested in LXD related projects.   I moderate  **[the LXD Sub-Reddit](https://www.reddit.com/r/LXD/)**.

But I have a project of my own that is for LXD and uses FRR, BGP, VXLAN and WireGuard to create an Internet Overlay system to interconnect LXD containers on multiple clouds/hosts and with multi-tenants.

**[CIAB Full Mesh VPN Internet Overlay](https://github.com/bmullan/CIAB.Full-Mesh.VPN.Wireguard.FRR.BGP.VXLAN.Internet.Overlay.Architecture)**

On Youtube are two Video's to help you understand and to install & configure the CIAB Full-Mesh VPN Internet Overlay Architecture using our Installation Guide see:

Video #1 (you can skip this one)

    Introduction to CIAB's Full-Mesh VPN Internet Overlay - approx 10 minutes long

Video #2

    Note: you can download the Installation Guide PDF (see above) if you want to follow it while watching Video #2

    CIAB Full-Mesh VPN Internet Overlay Installation Guide - Step-by-Step - approx 1 hr long

Video #2 will implement a 3 Node Full-Mesh VPN Internet Overlay using 3 Servers:

    1 - a Digital Ocean Cloud server located in the U.S.
    1 - a Hetzner Cloud server located in Germany
    1 - an LXD VM on one of my local servers traffic to/from this VM is limited by my local Wifi which is 200Mbps down and 20Mbps up.

At the end of Video #2 you will see Ping tests showing connectivity to/from ANY Node to/from ANY other Node's LXD Containers which are on private 10.x.x.x networks within each Node.

The 10.x.x.x network range is not Routeable over the Internet thus proving the success of the CIAB Full-Mesh VPN Internet Overlay in providing that universal inter-connectivity.

All Servers and LXD VM and LXD containers are running Ubuntu 20.04 LTS.

**Comment by Erik-Lamers1** [comment]
Very cool stuff. Definitely, something that can be useful for this project. In the end, I also want to make it possible to define remote endpoints for VNet-manager. That way you would only need to setup the machines on each end to be able to run LXC.
